app/utils/IsCountryUK.scala

```diff
@@ -0,0 +1,12 @@
+package utils
+
+import model.ConfirmableAddress
+
+object IsCountryUK {
+  val countryCheck: Option[ConfirmableAddress] => Boolean = (selectedAddress: Option[ConfirmableAddress]) => {
+    selectedAddress.fold(false)(
+      _.address.country.fold(true)(_.code == "GB")
+    )
```

Comment:
Same comment here as above - I find the above hard to follow.  The logic itself is a little tricky to express (since not all Nones get defaulted to the same value), but you might get further with splitting it up into separate functions?

Maybe something like:

```scala
def isConfirmableAddressInUk(confirmableAddress: ConfirmableAddress) = 
  confirmableAddress.address.country
    .map(_.code == "GB")
    .getOrElse(true)

selectedAddress.map(isConfirmableAddressInUK).getOrElse(false)
```

Reply:
again i think this is a per team basis.. our team is fine with folds.. I will change

Reply:
actually i disagree, i think it is readable <.<

Reply:
Definitely a subjective thing - I would appreciate the change in any case, though.  Of the two of us, at least one of us finds the fold-y version a bit difficult to parse. 

Happy to let @thomasrynnehmrc be the tie-breaker if you feel strongly against a change similar to the suggestion.

Reply:
I find the folds hard to read. Actually In this case I think I would use match :) but I prefer map getOrElse over fold.